### DISCUSSION TITLE
Welcome to tmf-hudl Discussions!

### DISCUSSION THREAD

**Discussion by remay** [remay/tmf-hudl#23] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Let us know how you got on.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by awooga** [comment]
How do you use this download?

just put on sdcard? Then what...my hudl seems to want a file from sd card? 
is the tsf custom Rom somewhere else? Or part of git code? 

Sorry old ...and don’t quite see how to progress

**Reply by remay** [reply]
Assuming you want to run this custom firmware on a Hudl 1, then I suggest you start here: https://github.com/remay/tmf-hudl/wiki/Flashing-from-an-SD-Card for detailed step-by-step instructions.   If you still have problems after trying to follow those instructions, then come back here and let me know which steps you need further help with.

**Reply by Robin1936** [reply]
Robin1936
I have been given a Hudl to fix, I should have read your story first, I also did a reset. I have tried to follow your procedure, but I have no success with it. I have downloaded the update but with errors. My total File size is1,202,240384, unzipped 579. I receive the following message, CAN NOT FOUND FIREWARE IMAGE OR IMAGE INVALID, UPDATE RKIMAGE COMPLETE. It would be much appreciated if you have any ideas.
Best regards.
Robin1736

**Reply by remay** [reply]
> Robin1936 I have been given a Hudl to fix, I should have read your story first, I also did a reset. I have tried to follow your procedure, but I have no success with it. I have downloaded the update but with errors. My total File size is1,202,240384, unzipped 579. I receive the following message, CAN NOT FOUND FIREWARE IMAGE OR IMAGE INVALID, UPDATE RKIMAGE COMPLETE. It would be much appreciated if you have any ideas. Best regards. Robin1736

Sorry for the delay in responding.  If the download goes correctly then the downloaded file (`hudl.20140424.153851.TMF-Custom-v1.0.img.zip`) should be 579 MB (607,191,872 bytes) according to the file propertirs in exxplorer, and the uzipped file should be 1.12 GB (1,204,166,968 bytes)

Try downloading it again.

**Reply by Robin1936** [reply]
Good Afternoon Robert.
Many thanks for your reply.
I reloaded the download as you suggested. There were a few changes, the size of the download was 1,204,166,968 bytes but the size on disc was 1,204,174,848 bytes. I do not know if this is significant, I still have the same result update.img is not recognised.
Best regards.
Robin1936

**Reply by remay** [reply]
Can you confirm the size of the SD card you are using and that it is formatted as a FAT32 filesystem?

So you downloaded and unzipped the image file and got a file called `hudl.20140424.153851.TMF-Custom-v1.0.img` with a size of 1,204,166,968 bytes (the size on disk is not important).  You re-named this file to be `update.img` and copied it to the root of the SD card?

Can you give the exact message that you see on the Hudl screen after selecting the 'update rkimage from external storage' option.

**Comment by awooga** [comment]
step 1: Download and unpack the TMF Custom ROM Firmware

Is that all thats at GIT project - or is the "TMF Custom ROM Firmware" somewhere else. Maybe obvious - but unclear. I downloaded the ZIP - is that the firmware?
Step 2: Copy the TMF Custom ROM Image onto a micro SDHC CardStep 3: Flash your Hudl from the SD cardStep 4: Perform the post-install steps###

Hi Robert,
thanks for quick response.

step 1: is this just the git project zip ?

Is that all thats at GIT project - or is the "TMF Custom ROM Firmware" somewhere else. Maybe obvious - but unclear. I downloaded the ZIP - is that the firmware?

step 2: just plain copy.

step 3: flash hudl... how ? if i try to upgrade hudl from sdcard is hunts for a file... on the sdcard which is not there... am i missing the TMF rom image?

thanks

brian

________________________________

**Reply by remay** [reply]
Click on the little black triangles on the left to expand each step of the instructions and see if you still have questions.

**Comment by awooga** [comment]
OMG!

I am such an idiot - there are hidden instructions... if you know the magic to find them.. lolz.

brian

i was at birth of www, my fault about trump - should have thrown tape of www into lake!

________________________________

**Reply by remay** [reply]
The fact that it wasn't obvious to you is interesting, as I want this to be as accessible to non-techies as possible.   I'm making a note to revise the design of thee pages.

**Comment by JavacHolt** [comment]
Tried this several times in the last couple of days, but it either doesn't find the IMG file or doesn't recognise it
![DSC_2549](https://user-images.githubusercontent.com/79976005/109796417-0b8c5800-7c10-11eb-9b6a-7a25b36e2bba.JPG)

**Reply by remay** [reply]
From what you've shown it finds the update.img file and validates it successfully.  It confirms that no change is needed to the parameter table.   It thinks it successfully updates the boot partition and then fails to update the system partition.

I've seen one other report of something similar, which after a prolonged investigation I came to the conclusion that the internal flash memory had somehow become set read-only.

You can clearly get the device to boot into recovery mode.  Does the device boot normally, or does it sit in a loop showing the "powered by android" logo for a long time and then going blank(rebooting) and repeating?

I'd be happy to work with you to try to confirm that you've got a device with failed flash - but unfortunately if that's the case I don't have a fix.

**Comment by JavacHolt** [comment]
Nope, slow to boot, but otherwise normal

**Reply by remay** [reply]
So you can boot normally to the "desktop" and run apps on the device?

In which case I haven't seen a failure to flash like that before.  If you're happy to download some different ".img" files that I can provide (most will be small) and try to install them using the same mechanism and return some log files to me then I'll see if we can work out what's going on.   I'll provide some instructions once you confirm you're happy to proceed.

**Comment by JavacHolt** [comment]
Yep, I can do that. Please bear in mind that I don't have or use a Windows
machine, just in case it gets to that.

**Reply by remay** [reply]
OK.  No Windows required.   Can we move this debugging session to email? It's somewhat easier to share the files we'll need to that way.  Drop me an email to D-29gckuvm@maildrop.cc (which is a temp email address only for this initial communication, I'll reply from my real email address).

**Reply by remay** [reply]
Sorry - I don't know if you tried, but there appears to be problems with delivery to that email address.  If you're happy to move to email let me know here and I'll find a better way for us to exchange email addresses.

**Comment by JavacHolt** [comment]
I did, but no worries. Whenever you're ready

**Comment by remay** [comment]
Thanks - can you try this one: nonila4483@onzmail.com
That should be good for an hour or so from now.

**Reply by remay** [reply]
I guess that didn't work either?   Let me create a new issue to track this and lets see if we can do it here on Github.

**Reply by remay** [reply]
I'd like to move the discussion to here please: #30 

If you add a comment to that issue you should start to get email notifications from it.

**Comment by AndrewChqng** [comment]
Hi, just want to ask for some advice on some stupid mistakes I did...
I accidently water damaged the Hudl after finally getting it running really well and in a moment of panic I didn't dry it properly so I corroded and killed the display control board when I turned it back on. I tried to clean up the corrosion but also scraped off contacts and my attempts to re-solder the pads did more damage then good.
Everything else works because I can see using the HDMI port but it's really hard to control and not much use in this state.
I would really like to fix it because I have so many childhood memories of using it and quite a bit of sentimental attachment.

I think this connector is gone for good...
![IMG_20210310_135003](https://user-images.githubusercontent.com/44511577/110641250-b02b0e80-81a9-11eb-8403-c8cf65ad2010.jpg)

I cleaned up the corrosion but no idea if it's alright. I feel like some of the components must be dead from how much corrosion I had to clean off
![IMG_20210310_134940](https://user-images.githubusercontent.com/44511577/110641123-85d95100-81a9-11eb-89e5-a480256262c0.jpg)

**Reply by remay** [reply]
It's difficult to see what you think you've broken (apart from the video connector to the main board) from teh photos.  If there was a lot of water damage then it looks like you'e done a good job cleaning it as I don't see it.

If you can see images on the HDMI output then that's a pretty good sign that most of the main board is OK.
And you imply that you can make touch input (even if difficult) which seems to suggest that those parts work.

I think you'll have a great deal of difficulty repairing that video ribbon cable now you have solder all over it.  I'm not sure about the connector on the main board - I don't think it's meant to opened, but it is possible to carefully flip the black part up and slide the ribbon in/out.   However I've broken the black part doing this in the past and that makes the connector unusable.

If I was youI'd be looking for a cheap 'for parts' Hudl on Ebay with a broken screen, and I'd salvage the main board with video connector from that.

Of course, if you think the video driver board is also broken, then you need a new screen/front case too as they are connected permanently by a multi-way ribbon cable:
![1615388993684](https://user-images.githubusercontent.com/83421/110650960-eae57480-81b2-11eb-970c-b679d3d2c973.jpg)

Here's a (broken) video cable connector on the main board with the remaining half black part flipped up:
![1615389219213](https://user-images.githubusercontent.com/83421/110651503-6810e980-81b3-11eb-9bcd-6639ab30a06e.jpg)


Rob.

**Comment by Burfotron1** [comment]
Hello 

I have followed the step by step guide on how to download and install flashing from a SD Card, When I insert the card into the Hudl and follow your help this shows below 

Recovery system v4.2.41 Update rkimage, finding update package UPDATE RKIMAGE find and check firmware can not found firmware image or invalid image update rkimage complete and also shows error on the screen with the yellow robot

Please can you let me know what I am doing wrong 

Thanks Amanda

**Reply by remay** [reply]
HI Amanda,

"find and check firmware can not found firmware image or invalid image". means that the Hudl is not finding the image on the SD Card, or that the image that it finds is somehow corrupt.

Check each of these steps carefully to make sure you have the right download and have processed it correctly once downloaded:
1. Download https://github.com/remay/tmf-hudl/releases/download/tmf-custom-rom-v1.0/hudl.20140424.153851.TMF-Custom-v1.0.img.zip 
2. Unzip it (you should get a file hudl.20140424.153851.TMF-Custom-v1.0.img (with size 1,204,166,968 bytes) 
3. rename the extracted file 'update.img'
4. put it on your SD card (in the root directory)
5. put the SD card into the Hudl
6. Boot the Hudl into recovery
7. select the 'update rkimge from external storage' option

If it still doesn't work let me know and we'll see how we can further help you work out what's going on.

Rob.

**Comment by Burfotron1** [comment]
Hello Rob

I have downloaded the link, but after unzipping this into another folder,
it shows this as the size 1,175,945kb and not (with size 1,204,166,968
bytes)

I deleted the - hudl.20140424.153851.TMF-Custom-v1.0.img- and changed it to
'update.img' and then copied it to the brand new micro sd card which I
checked and formatted to FAT32 and  which is also in an usb adaptor and
plugged into my laptop

I have just tried it again and it is still showing the same on the Hudl,
error etc

Please can you help

Thanking you
Kind regards
Amanda



<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Reply by remay** [reply]
I'd start by checking that what you're downloading is complete.

Download: https://github.com/remay/tmf-hudl/releases/download/tmf-custom-rom-v1.0/hudl.20140424.153851.TMF-Custom-v1.0.img.zip
Check MD5 of that is correct: `MD5 (hudl.20140424.153851.TMF-Custom-v1.0.img.zip) = 2da106b025060c9a6608d84d17132166`

What platform are you downloading on?

**Reply by remay** [reply]
1,175,945kb is probably the right size.

> I deleted the - hudl.20140424.153851.TMF-Custom-v1.0.img- and changed it to 'update.img'

No, dont delete `hudl.20140424.153851.TMF-Custom-v1.0.img`, as that's the file you need.  Just rename it to `update.img` and put it on the SD card.

**Reply by Burfotron1** [reply]
I am a bit new to this, learning as I go along, when you say platform, is it Windows 10 home 64 bit, which is on my Lenovo laptop

**Reply by Burfotron1** [reply]
I did not delete the file, I just renamed it by deleting the wording hudl.2010424.153851. etc and entered update.img and then copied it to the SD card

**Reply by remay** [reply]
Sorry, I'm not a native windows user.  But I *think* you can do the following:
1. Double click on the file `hudl.20140424.153851.TMF-Custom-v1.0.img.zip`, should leave you with a window showing a single file `hudl.20140424.153851.TMF-Custom-v1.0.img`
2. Copy (drag) that file to the SD card
3. on the copy on the SD card right-click and 'rename' to change the file on the SD card to `update.img`

Try that.

**Reply by Burfotron1** [reply]
I have just downloaded the link you have just sent me and I have unzipped it, but before it completed a box popped up to say The disc image file is corrupted ?

**Reply by remay** [reply]
> but before it completed a box popped up to say The disc image file is corrupted ?

That suggests your download is incomplete or corrupted.   Try again and check the MD5 checksum against the value I gave a few messages back.     For calculating the checksum from your downloaded file see: https://onthefencedevelopment.com/2017/08/15/windows-10-builtin-md5-checksum-calculator/ (many other similar references on the web)

Once you are sure you have a complete download then we'll try again.

**Reply by Burfotron1** [reply]
Just tried unzipping this download again and the same box is popping up - Couldn't Mount File - X The disc image file is corrupted

If  I am unable to update this Hudl with  your download, please can you let me know either way, so that I do not spend anymore time on trying to do this

Thanks

**Reply by remay** [reply]
> Just tried unzipping this download again and the same box is popping up - Couldn't Mount File - X The disc image file is corrupted

Are you double clicking the `.img' file?   That's the only way I can think that you'd get that message.

Let me come back to you later today when I have some time to pull a bit more information together for you to try to walk you through the process and to ensure we have the correct update.img on your SD card.

**Reply by remay** [reply]
OK, Try this link: https://we.tl/t-8fGMkH69O8 That should give you a downloaded file called 'update.img'.
If you right-click->properties you should see it having size 1,204,166,968 bytes:
![filesize](https://user-images.githubusercontent.com/83421/122918210-c9218f00-d356-11eb-8eaf-33235111103c.jpg)

Try putting that on your SD card and following the instructions from step 3: https://github.com/remay/tmf-hudl/wiki/Flashing-from-an-SD-Card

Let me know what happens.

**Comment by Burfotron1** [comment]
Hello

I have just clicked on the link you have provided and it shows me a new
page and it shows a spiral sign and this  - Yikes, that page can't be
found. Head back to wetransfer.com or cheer up our eyes with some nice
things we wrote

 The reason this does not work,  could be it be that the link you have
included,  you have highlighted the word THAT -  Try this link:
https://we.tl/t-8fGMkH69O8That

I have copied the link without the That on the end and it is now
downloading update.img

Fingers crossed this will now work

Thanks

**Reply by remay** [reply]
Good spot.  Original message edited to correct the link.

**Comment by Burfotron1** [comment]
This has now downloaded, but it is showing - This type of file can harm
your computer. Do you want to keep update.img anyway and tabs keep or
discard !!!

Just to check is this download safe to use, will it harm my computer/laptop

Please advise

Thanks

**Reply by remay** [reply]
It's safe. It's the unzipped and renamed contents of the .zip file that you appear to be having trouble downloading and/or expanding.

**Comment by Burfotron1** [comment]
Hello

This has now worked, thank you so much for your patience and help, this is
much appreciated

I have now a working Hudl

Thanks
Amanda

**Reply by remay** [reply]
Glad to hear that - I'd be interested in knowing what you were did wrong so that I can see if the instructions need clarifying anywhere.   The `update.img` file that you downloaded from WeTransfer should be *identical* to the `hudl.20140424.153851.TMF-Custom-v1.0.img` file that you extracted from the `.zip` file downloaded from the Releases page of Github.  (I generated it by downloading the file from Github, unziping it and renaming it).

Do you have any idea what you did wrong?  If not then don't worry, and I'm glad you have a working Hudl.

**Comment by Burfotron1** [comment]
Hello

As far as I know, I did nothing wrong, it just kept saying corrupted file ?

I downloaded the link, then extracted the file and before I copied it to
the sd card, I changed the name to img.file, but you did say in one of your
replies to copy it to the sd card and to change the name there, maybe that
is where I went wrong?

This was all new to me, learning as I was going along, but got there in the
end

Once again thank you for all your help

Kind regards
Amanda

**Comment by RayyanAnsari** [comment]
Hi everyone!
All the other forums about the Hudl seem to be dead, so I will ask my question here.
I've tried to flash the CWM Recovery linked here: https://www.modaco.com/topic/366202-clockworkmod-recovery-v6031-for-rkflashtool-built-by-hal2710/
I followed the exact instructions linked there, but the recovery still seems to be the stock system recovery....
Does anyone know why? Thanks!

**Reply by AndrewChqng** [reply]
Is this on Linux or Windows?

**Reply by RayyanAnsari** [reply]
This is on Ubuntu 20.04, using the `rkflashtool` from apt.

**Reply by remay** [reply]
What output do you get from
```
rkflashtool v
```
post the output.

We can do a little investigation together, but note that I have seen a couple of Hudls now where the Flash chip appears to have gone read-only.

You can try this:
```
rkflashtool r recovery > recovery.orig.img
rkflashtool w recovery < recovery.img      (where recovery.img is the file from the CWM-RECOVERY directory of the ZIP file you downloaded)
rkflashtool r recovery > recovery.cwm.img
cmp recovery.img recovery.cwm.img
```
The first line should read the contents of the recovery partition on the Huld into a file.
The second line should write the CWM image to the recovery partition
The 3rd line should read the new contents of the recovery partition into a new file
The 4th line compares what we think we wrote in the second line with what's actually read in the third line. 
If that last cmp is silent then you should be good;  otherwise we'll need to investigate further.

Rob.

**Reply by RayyanAnsari** [reply]
> We can do a little investigation together, but note that I have seen a couple of Hudls now where the Flash chip appears to have gone read-only.

I don't think my Hudl's flash chip has gone read only because I can install new apps, save files, etc.
I also know that my USB cable is good and rkflashtool works because I rooted my Hudl yesterday [by using rkflashtool to modify the system partition](https://www.earth.li/~noodles/blog/2020/12/rooting-tesco-hudl.html).

I'll do the commands later.

**Reply by remay** [reply]
I think if you do what I suggest above it'll all look OK, but after you successfully boot it'll revert to the stock recovery.

`/etc/install-recovery.sh` is run on every boot and the default looks like this:

```
#!/system/bin/sh
if ! applypatch -c MTD:recovery:16318464:f7887159742bae0e8c01afce5761d6c3a3f8701f; then
  log -t recovery "Installing new recovery image"
  applypatch -b /system/etc/recovery-resource.dat MTD:boot:9748480:d3cb32a5fcb45afe7a6eb6220f2cfcc02b3fef8e MTD:recovery f7887159742bae0e8c01afce5761d6c3a3f8701f 16318464 d3cb32a5fcb45afe7a6eb6220f2cfcc02b3fef8e:/system/recovery-from-boot.p
else
  log -t recovery "Recovery image already installed"
fi
```
and if I understand that correctly, then if the recovery partition looks like it has been changed then ~~both~~ the recovery ~~and boot~~ partitions get re-written.

Either remove/rename that file, or comment out the lines above (a number of utilities including this custom ROM add to the file to file taking advantage of the fact that it is run at boot with root permissions).

I had spotted this before, but hadn't got to looking at custom recovery programs and so hadn't documented it.

**Reply by RayyanAnsari** [reply]
Thanks, that worked!

**Reply by remay** [reply]
@WaluigiWare64 can you confirmed what worked?   Was that modifying/removing install-recovery.sh?

**Reply by RayyanAnsari** [reply]
Yes, I commented out each line in the install-recovery script.

**Comment by BigglesZ** [comment]
Hi
Does anyone have experience of this solution (or others) working with a Hudl2?
I've tried loading this update.img file but the file wasn't recognised. 
I've also tried the other recommended solutions but they's fail when loading from the app store the tablet locks up.
Any help would be much appreciated. Trying not to throw the thing away!
Thanks Andy

**Reply by remay** [reply]
This project is not compatible with Hudl2.     I am not aware of anyone ever having Rooted the Hudl2, which would be a prerequisite for a similar solution on the Hudl2.   I have had some reports that the sequence of steps documented [here](https://rob.themayfamily.me.uk/hudl/hudl-fix) also work to recover a Hudl2 that has been factory reset, but I don't have a Hudl2 device to check.  Rob.

**Reply by BigglesZ** [reply]
Hi rob
Thanks for responding and your advice on my new paperweight/tablet!
It was good to be able to rule out the change root option. As I was not knowledgeable enough to understand how this worked (just trying everything)
The recommended solution you pointed to I’d already tried unsuccessfully. All was good up to task 13 where you type "settings". My version of browser doesn’t show the icons on the right from the list of options.  So, I'm unable to get to the setting app that way.
I can navigate to the app store, but the tablet locks up when trying to install the settings link app.
With these options failing I think it's probably time to retire the tablet.
Your help has been appreciated, Andy

**Comment by Robin1936** [comment]
Hello again.

 

            The MicroSD Card is 32Gb and is formatted to FAT32

 

            Recovery system v4.2.4.1

 

           -Update rkimage-

           Finding update package...

           ===UPDATE IMAGE===

          Find and check Firmware...

          Can not found firmware image or invalid image.

             Update rkimage complete.

 

          I hope this info. helps. (not sure what send to root of SD card 
means)

          Best regards

          Robin Shopland.

 

----- Original Message -----
From: Robert May ***@***.***>
Reply-To: remay/tmf-hudl 
***@***.***>
To: remay/tmf-hudl ***@***.***>
Cc: Robin1936 ***@***.***>, Comment 
***@***.***>
Sent: 16/02/2023 14:29:25
Subject: Re: [remay/tmf-hudl] Welcome to tmf-hudl Discussions! (Discussion #23)



Can you confirm the size of the SD card you are using and that it is formatted 
as a FAT32 filesystem?
So you downloaded and unzipped the image file and got a file called 
hudl.20140424.153851.TMF-Custom-v1.0.img with a size of 1,204,166,968 bytes 
(the size on disk is not important). You re-named this file to be update.img 
and copied it to the root of the SD card?
Can you give the exact message that you see on the Hudl screen after selecting 
the 'update rkimage from external storage' option.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: <remay/tmf-hudl/repo-
***@***.***>

**Reply by remay** [reply]
That means that the update file was not found on the SD card, or it was found and didn't have the correct content.
As we have confirmed that the file size is correct, it is unlikley that the content is incorrect, so let's focus on the file name.

By default Explorer hides known file name extensions.   Can you do the following:
1. In explorer browse to the SD card so that you can see the file in the main window
2. On the Toolbr/Ribbon at the top click on 'View' and the towards the right find the check box with 'File name extensions' next to it, and check that checkbox.
3. Now make sure the filename is `update.img` (and not `update.img.img`).  If it's the latter then rename to `update.img` and retry.  Else come back and let me know.

**Comment by Robin1936** [comment]
Hello Robert.

 

             Great success, you have solved my problem

 

             Many many thanks.

 

             Very happy best regards.

 

            Robin Shopland.

 

----- Original Message -----
From: Robert May ***@***.***>
Reply-To: remay/tmf-hudl 
***@***.***>
To: remay/tmf-hudl ***@***.***>
Cc: Robin1936 ***@***.***>, Comment 
***@***.***>
Sent: 17/02/2023 17:33:23
Subject: Re: [remay/tmf-hudl] Welcome to tmf-hudl Discussions! (Discussion #23)



That means that the update file was not found on the SD card, or it was found 
and 
didn't have the correct content.
As we have confirmed that the file size is correct, it is unlikley that the content is incorrect, so let's 
focus on the file name.
By default Explorer hides known file name extensions. Can you do the following:
In explorer browse to the SD card so that you can see the file in the main 
window 
On the Toolbr/Ribbon at the top click on 'View' and the towards the right find 
the check box with 'File name extensions' next to it, and check that checkbox. 
Now make sure the filename is update.img (and not update.img.img). If it's the latter then rename to update.img and retry. Else come back and let me know.

**Reply by remay** [reply]
> Hello Robert. Great success, you have solved my problem Many many thanks. Very happy best regards. Robin Shopland.

Really glad to have got to the bottom of that one.  I've seen a few other people reporting something similar and have ended up solving it in different ways for them as I never got to the root cause.  I'm going to add a little bit of further information to the instructions to try to avoid others falling foul of this.
